ext/store/bigquery/client.go

```diff
@@ -35,29 +39,127 @@ func NewClient(ctx context.Context, svcAccount string) (*BqClient, error) {
 		return nil, errors.InternalError(store, "failed to create BQ client", err)
 	}
 
-	return &BqClient{bq: c}, nil
+	return &BqClient{c}, nil
 }
 
 func (c *BqClient) DatasetHandleFrom(ds Dataset) ResourceHandle {
-	dsHandle := c.bq.DatasetInProject(ds.Project, ds.DatasetName)
+	dsHandle := c.DatasetInProject(ds.Project, ds.DatasetName)
 	return NewDatasetHandle(dsHandle)
 }
 
 func (c *BqClient) TableHandleFrom(ds Dataset, name string) TableResourceHandle {
-	t := c.bq.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
+	t := c.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
 	return NewTableHandle(t)
 }
 
 func (c *BqClient) ExternalTableHandleFrom(ds Dataset, name string) ResourceHandle {
-	t := c.bq.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
+	t := c.DatasetInProject(ds.Project, ds.DatasetName).Table(name)
 	return NewExternalTableHandle(t)
 }
 
+func (c *BqClient) BulkGetDDLView(ctx context.Context, dataset Dataset, names []string) (map[job.ResourceURN]string, error) {
+	queryContent := buildGetDDLQuery(dataset.Project, dataset.DatasetName, names...)
+	queryStatement := c.Client.Query(queryContent)
+	rowIterator, err := queryStatement.Read(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	urnToDDL := map[job.ResourceURN]string{}
+	var errorMessages []string
+	for {
+		var values []bigquery.Value
+		if err := rowIterator.Next(&values); err != nil {
+			if errors.Is(err, iterator.Done) {
+				break
+			}
+
+			errorMessages = append(errorMessages, err.Error())
+			continue
+		}
+
+		if len(values) == 0 {
+			continue
+		}
+
+		name, ddl, err := getViewDDL(values)
+		if err != nil {
+			errorMessages = append(errorMessages, err.Error())
+			continue
+		}
+
+		resourceURN := job.ResourceURN("bigquery://" + fmt.Sprintf("%s:%s.%s", dataset.Project, dataset.DatasetName, name))
```

Comment:
We should not be using resourceURN from job, seems like unnecessary coupling. This can lead to unnecessary transitive dependency between modules.

We can use a better model as we are going to split the fields again.

Reply:
Would you mean it's better to create their own ResourceURN type within this package? Or we could have dedicated package just to serve this model?

I'm more inclined to put ResourceURN on resource bounded context, as it has closer relation with this BC

Reply:
Please refer to this article: https://martinfowler.com/bliki/BoundedContext.html
2 objects sharing same name does not mean they are same object. It depends on the responsibility of the objects if there are same or not.

Let's have a discuss if it is not clear.